src/util/logger/mod.rs

```diff
@@ -190,6 +190,32 @@ where
     }
 }
 
+struct LogAndFuse<D>(D);
+
+impl<D> Drain for LogAndFuse<D>
+where
+    D: Drain,
+    <D as Drain>::Err: std::fmt::Display,
+{
+    type Ok = ();
+    type Err = slog::Never;
+
+    fn log(&self, record: &Record<'_>, values: &OwnedKVList) -> Result<Self::Ok, Self::Err> {
+        if let Err(e) = self.0.log(record, values) {
+            let fatal_drainer = Mutex::new(term_drainer()).ignore_res();
+            fatal_drainer.log(record, values).unwrap();
+            let fatal_logger = slog::Logger::root(fatal_drainer, slog_o!());
+            slog::slog_crit!(
+                fatal_logger,
+                "logger encountered error, TiKV cannot continue working";
+                "err" => %e,
```

Comment:
Seems we print this error twice? Maybe only once is needed.

Reply:
@Hoverbear you mean the error will be printed in the `panic` below? I'd remove the `e` from the panic if this is indeed duplicated, since the panic message could be lost.

Reply:
Ok! 👍